File: descripe/imapp_modisl2/src/leocat/src/MOD_PRAlg29/chk_spatial2.c
/*
************************************************************************
!F77 

 !Description:

   Routine for checking spatial variability of ocean scenes over
   a 3X3 pixel region.

!Input parameters:
   none

   Inputs stored in structure variable 'pxin' of type "pixel_in" defined
   in pixel.h. 

!Output Parameters:
   none

   Outputs stored in structure variable 'rg_var' of type "regional_var"
   in pixel.h.

!Revision History:

   06/04 Collection 5  R. Frey:
   11/07 Converted to C  R. Frey

!Team-unique Header:

!References and Credits:
   See Cloud Mask ATBD-MOD-35.

!END
***********************************************************************/

/* Includes */

#include <stdio.h>
#include "pixel.h"


void chk_spatial2()


{

/*  Declarations */

    extern void get_regdif(int);
    extern int spatial_var();

    int band;
    int var;

/**********************************************************************/

//  Get brightness temperature differences between pixel of interest
//  and the 8 surrounding it.

    band = 31;
    (void) get_regdif(band);

//  Check variation in the region.
    var = spatial_var();
        
    return;

}

File: descripe/imapp_modisl2/src/cloudmask/src/chk_spatial2.f
      subroutine chk_spatial2(indat,kele,npix)


c---------------------------------------------------------------------
C!F77 
c
c!Description:
c
c     Routine for checking spatial variability of ocean scenes over
c     a 3X3 pixel region.
c
c!Input parameters:
c indat         Array containing nlcntx lines of data
c kele          Current granule element number being processed
c
c!Output Parameters:
c npix          Number pixels (max of 8) which satisfy variability test
c
c!Revision History:
c $Id: chk_spatial2.f,v 1.1.2.1 2004/05/27 15:55:24 raf Exp $
c
c!Team-unique Header:
c
c!References and Credits:
c See Cloud Mask ATBD-MOD-06.
c
c!END
c---------------------------------------------------------------------

      include 'global.inc'

c     Scalar arguments
      integer kele,npix

c     Array arguments
      real indat(npixel,nlcntx,inband)

c     Local scalars
      integer varslt,h_output,debug

c     Local Arrays
      real diff(var_band,8),rgdata(nlcntx,necntx,var_band)

c     External subroutines
      external get_regdif,spatial_var

c     Common statement for debug purposes
      common / bug / debug, h_output

c---------------------------------------------------------------------

c     Get brightness temperature differences between pixel of interest
c     and the 8 surrounding it.
      call get_regdif(indat,kele,rgdata,diff)

c     Check variation in the region.
      call spatial_var(diff,npix,varslt)
        
      return
      end
